Modules/ITS/src/ITSTrackTask.cxx

```diff
@@ -0,0 +1,222 @@
+    mNTracks = 0;
+    mNRofs = 0;
+
+    hClusterUsage->SetBinContent(1, 1. * mNClustersInTracks / mNClusters);
+    mNClustersInTracks = 0;
+    mNClusters = 0;
+  }
+}
+
+void ITSTrackTask::endOfCycle()
+{
+
+  std::ifstream runNumberFile("/home/its/QC/workdir/infiles/RunNumber.dat"); //catching ITS run number in commissioning
+  if (runNumberFile) {
+
+    std::string runNumber;
+    runNumberFile >> runNumber;
+    if (runNumber != mRunNumber) {
+      for (unsigned int iObj = 0; iObj < mPublishedObjects.size(); iObj++)
+        getObjectsManager()->addMetadata(mPublishedObjects.at(iObj)->GetName(), "Run", runNumber);
+      mRunNumber = runNumber;
+    }
+    QcInfoLogger::GetInstance() << "endOfCycle" << AliceO2::InfoLogger::InfoLogger::endm;
+  }
+}
+
+void ITSTrackTask::endOfActivity(Activity& /*activity*/)
+{
+  QcInfoLogger::GetInstance() << "endOfActivity" << AliceO2::InfoLogger::InfoLogger::endm;
+}
+
+void ITSTrackTask::reset()
+{
+  QcInfoLogger::GetInstance() << "Resetting the histogram" << AliceO2::InfoLogger::InfoLogger::endm;
+  hAngularDistribution->Reset();
+  hNClusters->Reset();
+  hTrackPhi->Reset();
+  hTrackEta->Reset();
+  hOccupancyROF->Reset();
+  hClusterUsage->Reset();
+}
+
+void ITSTrackTask::createAllHistos()
+{
+
+  hAngularDistribution = new TH2D("AngularDistribution", "AngularDistribution", 30, -1.5, 1.5, 60, 0, TMath::TwoPi());
+  hAngularDistribution->SetTitle("AngularDistribution");
+  addObject(hAngularDistribution);
+  formatAxes(hAngularDistribution, "#eta", "#phi", 1, 1.10);
+  hAngularDistribution->SetStats(0);
+
+  hNClusters = new TH1D("NClusters", "NClusters", 100, 0, 100);
+  hNClusters->SetTitle("hNClusters");
+  addObject(hNClusters);
+  formatAxes(hNClusters, "Number of clusters per Track", "Counts", 1, 1.10);
+
+  hTrackEta = new TH1D("EtaDistribution", "EtaDistribution", 30, -1.5, 1.5);
+  hTrackEta->SetTitle("Eta Distribution of tracks");
+  addObject(hTrackEta);
+  formatAxes(hTrackEta, "#eta", "counts", 1, 1.10);
+
+  hTrackPhi = new TH1D("PhiDistribution", "PhiDistribution", 60, 0, TMath::TwoPi());
+  hTrackPhi->SetTitle("Phi Distribution of tracks");
+  addObject(hTrackPhi);
+  formatAxes(hTrackPhi, "#phi", "counts", 1, 1.10);
+
+  hOccupancyROF = new TH1D("OccupancyROF", "OccupancyROF", 1, 0, 1);
+  hOccupancyROF->SetTitle("Track occupancy in ROF");
+  addObject(hOccupancyROF);
+  formatAxes(hOccupancyROF, "", "nTracks/ROF", 1, 1.10);
+
+  hClusterUsage = new TH1D("ClusterUsage", "ClusterUsage", 1, 0, 1);
+  hClusterUsage->SetTitle("Fraction of clusters used in tracking");
+  addObject(hClusterUsage);
+  formatAxes(hClusterUsage, "", "nCluster in track / Total cluster", 1, 1.10);
+}
+
+void ITSTrackTask::getEnableLayers()
+{
+  std::ifstream configFile("Config/ConfigLayers.dat"); //passing enabled layers into QC for ITS commissioning
+  for (int ilayer = 0; ilayer < NLayer; ilayer++) {
+    configFile >> mEnableLayers[ilayer];
+    if (mEnableLayers[ilayer])
```

Comment:
we recommend to "use curly braces even when the body of the statement fits on one line"

Reply:
modified.